### PR TITLE
OCPBUGS-37056,OCPBUGS-37439,OCPBUGS-37439: Downstream Merge 14th August 2024

### PR DESCRIPTION
**Pull request by tssurya**
https://github.com/ovn-org/ovn-kubernetes/pull/4624 looks like the fix we need to ensure we don't flush conntrack entries.. so trying this again
https://github.com/ovn-org/ovn-kubernetes/pull/4631 was required to solve the delete pod alert

### CODE REVIEW

**Comment by tssurya** [comment]
/payload-job-with-prs 4.18 ci blocking openshift/api#1997
/payload-job-with-prs 4.18 nightly blocking openshift/api#1997

**Comment by tssurya** [comment]
/payload-job-with-prs 4.18 ci blocking https://github.com/openshift/api/pull/1997

**Comment by tssurya** [comment]
/payload-job-with-prs 4.18 ci blocking openshift/api#1997
/payload-job-with-prs 4.18 nightly blocking openshift/api#1997

**Comment by tssurya** [comment]
/payload-with-prs 4.18 ci blocking https://github.com/openshift/api/pull/1997

**Comment by tssurya** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.18-e2e-metal-ipi-ovn-ipv6 openshift/api#1997

File: go-controller/pkg/ovn/secondary_layer3_network_controller_test.go [hunk 1]
```diff
@@ -453,12 +453,8 @@ func expectedGWEntities(nodeName string, netInfo util.NetInfo, gwConfig util.L3G
 		expectedGatewayChassis(nodeName, netInfo, gwConfig),
 		expectedStaticMACBinding(gwRouterName, nextHopMasqueradeIP()),
 	)
-	for _, entity := range expectedExternalSwitchAndLSPs(netInfo, gwConfig, nodeName) {
-		expectedEntities = append(expectedEntities, entity)
-	}
-	for _, entity := range expectedJoinSwitchAndLSPs(netInfo, nodeName) {
-		expectedEntities = append(expectedEntities, entity)
-	}
+	expectedEntities = append(expectedEntities, expectedExternalSwitchAndLSPs(netInfo, gwConfig, nodeName))
```
Comment (tssurya) [inline go-controller/pkg/ovn/secondary_layer3_network_controller_test.go:456]: oopsy missing ... which is fixed in the original upstream PR will need to repull

**Comment by trozet** [comment]
tech preview job shows
```
OVNKubernetesNodePodDeleteError has no test data, this alert appears new and should not be firing}
```

we need to fix this this issue either in the test or upstream before we can merge

**Comment by tssurya** [comment]
> tech preview job shows
> 
> ```
> OVNKubernetesNodePodDeleteError has no test data, this alert appears new and should not be firing}
> ```
> 
> we need to fix this this issue either in the test or upstream before we can merge

oh this error was not present  yesterday! its one of the new commits I think. See my previous run yesterday: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2259/pull-ci-openshift-ovn-kubernetes-master-e2e-gcp-ovn-techpreview/1823905939745411072

**Comment by tssurya** [comment]
I have pulled in https://github.com/ovn-org/ovn-kubernetes/pull/4631/commits to see if this fixes it

**Comment by tssurya** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-ci-4.18-e2e-gcp-ovn-techpreview openshift/origin#29012

**Comment by tssurya** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-ci-4.18-e2e-gcp-ovn-techpreview openshift/origin#29012

**Comment by tssurya** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview openshift/origin#29012

**Comment by tssurya** [comment]
/payload-with-prs 4.18 nightly blocking openshift/api#1997,openshift/origin#29012

**Comment by tssurya** [comment]
/payload-with-prs 4.18 ci blocking openshift/api#1997,openshift/origin#29012

**Comment by tssurya** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview openshift/origin#29012

**Comment by tssurya** [comment]
/payload-with-prs 4.18 ci blocking openshift/api#1997, openshift/origin#29012

**Comment by tssurya** [comment]
/payload-with-prs 4.18 ci blocking openshift/api#1997 openshift/origin#29012

**Comment by tssurya** [comment]
/payload-with-prs 4.18 nightly blocking openshift/api#1997 openshift/origin#29012

**Comment by trozet** [comment]
fixed the other reason that CNI DEL was erroring out and causing tech preview job to fail:
https://github.com/ovn-org/ovn-kubernetes/pull/4631

**Comment by martinkennelly** [comment]
/retitle OCPBUGS-37056,OCPBUGS-37439,OCPBUGS-37439: Downstream Merge 14th August 2024

**Comment by tssurya** [comment]
I am very confused, why are all the jobs retriggered on this PR? We had already run all of them...

**Comment by tssurya** [comment]
I also don't understand why 4.16->4.17 jobs are being triggered here :) I think its because I opened a sync against 4.17 as well?

**Pull request merged**